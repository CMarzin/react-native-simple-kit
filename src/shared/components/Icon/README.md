# Icon ðŸ’ 

A generic component for SVG icons.

### API
* name: `'backButton'`
* actived?: `boolean`
* size?: `number`
* defaultActivedColor?: `string`
* defaultColor?: `string`
* style?: `Stylesheet | number`

### Example
```js
import React from 'react'
import Icon from '@components/Icon'

const MyComponent = (): React$Element<any> => (
  <Icon
    name="google"
    defaultColor="red"
    size={22}
  />
)

export default MyComponent
```

## Changelog
#### Wednesday, November 15th
- Initial release. We didn't track changes before this version.
