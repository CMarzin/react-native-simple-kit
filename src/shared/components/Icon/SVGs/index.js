/* @flow */

export { default as Back } from './Back'
