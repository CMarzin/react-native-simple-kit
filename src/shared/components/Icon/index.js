/* @flow */

import React, { createElement } from 'react'
import PropTypes from 'prop-types'
import { SafeAreaView } from 'react-native'

import { NEUTRAL_COLOR_50, BRAND_COLOR_GREEN } from '@theme/colors'
import * as SVGs from './SVGs'

type Props = {
  actived?: boolean,
  defaultactivedColor?: string,
  defaultColor?: string,
  name: string,
  size: number,
  style?: ?StyleSheet,
}

const Icon = (props: Props): React$Element<any> => {
  const { name, style, size, actived, defaultColor, defaultactivedColor } = props
  const color = actived
    ? defaultactivedColor || BRAND_COLOR_GREEN
    : defaultColor || NEUTRAL_COLOR_50

  let icon
  switch (name) {
    case 'backButton': icon = SVGs.Back
      break
    default: {
      icon = createElement(SafeAreaView, { ...props })
      throw new Error(`Invalid SVG name: ${name}`)
    }
  }

  return (
    <SafeAreaView {...props} style={style}>
      {createElement(icon, { color, size })}
    </SafeAreaView>
  )
}

Icon.propTypes = {
  actived: PropTypes.bool,
  defaultactivedColor: PropTypes.string,
  defaultColor: PropTypes.string,
  name: PropTypes.string.isRequired,
  size: PropTypes.number.isRequired,
  style: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.object,
  ]), // eslint-disable-line react/forbid-prop-types
}

Icon.defaultProps = {
  actived: false,
  defaultactivedColor: BRAND_COLOR_GREEN,
  defaultColor: NEUTRAL_COLOR_50,
  style: undefined,
}

export default Icon
