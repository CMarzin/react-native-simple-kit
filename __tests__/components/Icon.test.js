import '@mocks/SVG'
import React from 'react'
import TestRenderer from 'react-test-renderer'
import Icon from '@components/Icon'

describe('Generic component: <Icon />', () => {
  it('renders correctly', () => {
    const component = TestRenderer.create(
      <Icon
        name="backButton"
        size={15}
      />,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders actived version correctly', () => {
    const component = TestRenderer.create(
      <Icon
        actived
        name="backButton"
        size={15}
      />,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders with actived version with custom color correctly', () => {
    const component = TestRenderer.create(
      <Icon
        actived
        defaultactivedColor="#5AADEB"
        name="backButton"
        size={15}
      />,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders with defaultColor correctly', () => {
    const component = TestRenderer.create(
      <Icon
        defaultColor="#333333"
        name="backButton"
        size={15}
      />,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
