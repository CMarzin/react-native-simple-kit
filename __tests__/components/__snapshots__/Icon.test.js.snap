// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generic component: <Icon /> renders actived version correctly 1`] = `
<RCTSafeAreaView
  actived={true}
  defaultColor="#b8b8b8"
  defaultactivedColor="#53C000"
  name="backButton"
  size={15}
  style={undefined}
>
  <Svg />
</RCTSafeAreaView>
`;

exports[`Generic component: <Icon /> renders correctly 1`] = `
<RCTSafeAreaView
  actived={false}
  defaultColor="#b8b8b8"
  defaultactivedColor="#53C000"
  name="backButton"
  size={15}
  style={undefined}
>
  <Svg />
</RCTSafeAreaView>
`;

exports[`Generic component: <Icon /> renders with actived version with custom color correctly 1`] = `
<RCTSafeAreaView
  actived={true}
  defaultColor="#b8b8b8"
  defaultactivedColor="#5AADEB"
  name="backButton"
  size={15}
  style={undefined}
>
  <Svg />
</RCTSafeAreaView>
`;

exports[`Generic component: <Icon /> renders with defaultColor correctly 1`] = `
<RCTSafeAreaView
  actived={false}
  defaultColor="#333333"
  defaultactivedColor="#53C000"
  name="backButton"
  size={15}
  style={undefined}
>
  <Svg />
</RCTSafeAreaView>
`;
